### PR TITLE
Added setup.py and converted this into a proper module.

### PR DESCRIPTION
**Pull request by jgrassler**
This pull requests turns the primitives repository into a proper python module that can be installed using `pip install`.

### CODE REVIEW

File: cloudcix_primitives/utils.py [hunk 1]
```diff
@@ -75,41 +82,46 @@ def load_pod_config(config_file=None, prefix=4000) -> Dict[str, Any]:
...
+    config = None
+
     # Load config from config_file
     try:
         with Path(config_file).open('r') as file:
-            config['file'] = json.load(file)
+            config = json.load(file)
     except OSError as e:
-            return False, config_data, messages[prefix + 11] + e.__str__()
+            return False, config_data, ("%d: " % (prefix+11)) + messages[11] + e.__str__()
```
Comment (NaveenSidhartha) [inline cloudcix_primitives/utils.py:95]:
```suggestion
            return False, config_data, f'{prefix+11}: {messages[11]} {e.__str__()}'
```

f' strings are way simpler and clean to use.

File: cloudcix_primitives/utils.py [hunk 2]
```diff
@@ -75,41 +82,46 @@ def load_pod_config(config_file=None, prefix=4000) -> Dict[str, Any]:
...
     except Exception as e:
-            return False, config_data, messages[prefix + 12] + e.__str__()
+            return False, config_data, ("%d: " % (prefix+12)) + messages[12] + e.__str__()
```
Comment (NaveenSidhartha) [inline cloudcix_primitives/utils.py:97]:
```suggestion
            return False, config_data, f'{prefix+12}: {messages[12]} {e.__str__()}'
```

File: cloudcix_primitives/utils.py [hunk 3]
```diff
@@ -75,41 +82,46 @@ def load_pod_config(config_file=None, prefix=4000) -> Dict[str, Any]:
...
 
     config_data['raw'] = config
 
     # Get the ipv6_subnet from config_file
     ipv6_subnet = config.get('ipv6_subnet', None)
     if ipv6_subnet is None:
-        return False, config_data, ("%d: " % prefix + 13) + messages[13]
+        return False, config_data, ("%d: " % (prefix+13)) + messages[13]
```
Comment (NaveenSidhartha) [inline cloudcix_primitives/utils.py:104]:
```suggestion
        return False, config_data, f'{prefix+13}: {messages[13]} {e.__str__()}'
```

File: cloudcix_primitives/utils.py [hunk 4]
```diff
@@ -75,41 +82,46 @@ def load_pod_config(config_file=None, prefix=4000) -> Dict[str, Any]:
...
+    podnet_b = f'{ipv6_subnet.split("/")[0]}10:0:3'
+
+    config_data['processed']['podnet_a'] = podnet_a
+    config_data['processed']['podnet_b'] = podnet_b
 
     # Get `podnet_a_enabled` and `podnet_b_enabled`
     podnet_a_enabled = config.get('podnet_a_enabled', None)
     if podnet_a_enabled is None:
-        return False, config_data, ("%d: " % prefix + 15) + messages[15]
+        return False, config_data, ("%d: " % (prefix+15)) + messages[15]
```
Comment (NaveenSidhartha) [inline cloudcix_primitives/utils.py:121]:
```suggestion
        return False, config_data, f'{prefix+15}: {messages[15]} {e.__str__()}'
```

File: cloudcix_primitives/utils.py [hunk 5]
```diff
@@ -75,41 +82,46 @@ def load_pod_config(config_file=None, prefix=4000) -> Dict[str, Any]:
...
     podnet_b_enabled = config.get('podnet_b_enabled', None)
     if podnet_a_enabled is None:
-        return False, config_data, ("%d: " % prefix + 16) + messages[16]
+        return False, config_data, ("%d: " % (prefix+16)) + messages[16]
```
Comment (NaveenSidhartha) [inline cloudcix_primitives/utils.py:124]:
```suggestion
        return False, config_data, f'{prefix+16}: {messages[16]} {e.__str__()}'
```

File: cloudcix_primitives/utils.py [hunk 6]
```diff
@@ -119,16 +131,16 @@ def load_pod_config(config_file=None, prefix=4000) -> Dict[str, Any]:
         enabled = podnet_b
         disabled = podnet_a
     elif podnet_a_enabled is True and podnet_b_enabled is True:
-        return False, config_data, ("%d: " % prefix + 17) + messages[17]
+        return False, config_data, ("%d: " % (prefix+17)) + messages[17]
```
Comment (NaveenSidhartha) [inline cloudcix_primitives/utils.py:134]:
```suggestion
        return False, config_data, f'{prefix+17}: {messages[17]} {e.__str__()}'
```

File: cloudcix_primitives/utils.py [hunk 7]
```diff
@@ -119,16 +131,16 @@ def load_pod_config(config_file=None, prefix=4000) -> Dict[str, Any]:
...
     elif podnet_a_enabled is False and podnet_b_enabled is False:
-        return False, config_data, ("%d: " % prefix + 18) + messages[18]
+        return False, config_data, ("%d: " % (prefix+18)) + messages[18]
```
Comment (NaveenSidhartha) [inline cloudcix_primitives/utils.py:136]:
```suggestion
        return False, config_data, f'{prefix+18}: {messages[18]} {e.__str__()}'
```

File: cloudcix_primitives/utils.py [hunk 8]
```diff
@@ -119,16 +131,16 @@ def load_pod_config(config_file=None, prefix=4000) -> Dict[str, Any]:
...
     else:
-        return False, config_data, ("%d: " % prefix + 19) + messages[19]
+        return False, config_data, ("%d: " % (prefix+19)) + messages[19]
```
Comment (NaveenSidhartha) [inline cloudcix_primitives/utils.py:138]:
```suggestion
        return False, config_data, f'{prefix+19}: {messages[19]} {e.__str__()}'
```

File: cloudcix_primitives/utils.py [hunk 9]
```diff
@@ -119,16 +131,16 @@ def load_pod_config(config_file=None, prefix=4000) -> Dict[str, Any]:
...
 
     config_data['processed']['enabled'] = enabled
     config_data['processed']['disabled'] = disabled
 
-    return True, config_data, ("%d: " % prefix + 10) + messages[10]
+    return True, config_data, ("%d: " % (prefix+10)) + messages[10]
```
Comment (NaveenSidhartha) [inline cloudcix_primitives/utils.py:143]:
```suggestion
    return True, config_data, f'{prefix+10}: {messages[10]} {e.__str__()}'
```

File: cloudcix_primitives/utils.py [hunk 10]
```diff
@@ -206,35 +229,88 @@ def __init__(config_file, podnet_node, enabled, payload_channels, successful_pay
...
+                    context.append('        ' + payload['rcc_return']['payload_message'])
+                    context.append(f'      %s: ' % self.payload_channels['payload_error'] )
+                    context.append('        ' + payload['rcc_return']['payload_error'])
+            context.append("")
+            context.append("")
+        return "\n".join(context)
 
     def _format_channel_error(self, rcc_return, msg):
-        msg = msg + "channel_code: %s\nchannel_message: %s\nchannel_error: %s" % (
+        msg = msg + "channel_code: %s\nchannel_message: %s\nchannel_error: %s\n\n" % (
```
Comment (NaveenSidhartha) [inline cloudcix_primitives/utils.py:313]:
```suggestion
        msg = msg + f"channel_code: {rcc_return['channel_code']}\nchannel_message: {rcc_return['channel_message']}\nchannel_error: rcc_return['channel_error']\n\n" % (
```

File: cloudcix_primitives/utils.py [hunk 11]
```diff
@@ -206,35 +229,88 @@ def __init__(config_file, podnet_node, enabled, payload_channels, successful_pay
...
             rcc_return['channel_code'],
```
Comment (NaveenSidhartha) [inline cloudcix_primitives/utils.py:314]:
```suggestion

```

File: cloudcix_primitives/utils.py [hunk 12]
```diff
@@ -206,35 +229,88 @@ def __init__(config_file, podnet_node, enabled, payload_channels, successful_pay
...
             rcc_return['channel_error'],
```
Comment (NaveenSidhartha) [inline cloudcix_primitives/utils.py:315]:
```suggestion
```

File: cloudcix_primitives/utils.py [hunk 13]
```diff
@@ -206,35 +229,88 @@ def __init__(config_file, podnet_node, enabled, payload_channels, successful_pay
...
             rcc_return['channel_message']
```
Comment (NaveenSidhartha) [inline cloudcix_primitives/utils.py:316]: Also remove bracket in line 317

File: cloudcix_primitives/utils.py [hunk 14]
```diff
@@ -243,7 +319,7 @@ def _format_channel_error(self, rcc_return, msg):
         return msg
 
     def _format_payload_error(self, rcc_return, msg):
-        msg = msg + "channel_code: %s\nchannel_message: %s\nchannel_error: %s" % (
+        msg = msg + "payload code: %s\n%s: %s\n%s: %s\n\n" % (
```
Comment (NaveenSidhartha) [inline cloudcix_primitives/utils.py:322]: here as well use f'strings

**Review by NaveenSidhartha — DISMISSED** [review]

File: MANIFEST.in [hunk 15]
```diff
@@ -0,0 +1,5 @@
+include README.md
+include LICENSE
+include requirements.txt
```
[the remark this answers is not in the document]
Comment (maria-walsh) [reply MANIFEST.in:3]: @NaveenSidhartha All these files are in the project repo in the same directory as this file

[the remark this answers is not in the document]
Comment (NaveenSidhartha) [reply MANIFEST.in:3]: oh i was looking in Filechanged.

**Pull request merged**